### ISSUE TITLE
Creating a mock with multiple traits fails if traits share a name

### ISSUE THREAD

**Issue by discosultan** [asomers/mockall#600]
In mockall = "0.13":

```rs
use mockall::mock;

mod a {
    pub trait A {
        fn foo(&self);
    }
}

mod b {
    pub trait A {
        fn bar(&self);
    }
}

mock! {
    // Structure to mock
    C {}
    // First trait to implement on C
    impl a::A for C {
        fn foo(&self);
    }
    // Second trait to implement on C
    impl b::A for C {
        fn bar(&self);
    }
}
```

fails with:

```
error[E0428]: the name `__mock_MockC_A` is defined multiple times
  --> src/lib.rs:15:1
   |
15 | mock! {
   | ^^^^^
   | |
   | `__mock_MockC_A` redefined here
   | previous definition of the module `__mock_MockC_A` here
   |
   = note: `__mock_MockC_A` must be defined only once in the type namespace of this module
   = note: this error originates in the macro `mock` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by asomers** [comment]
Ahh yes.  That's bound to cause problems. I bet if you expand the code, you'll find more than just those two identically named structs, too.  This limitation is unlikely to be fixed.  But if it's really important to you, you could take a stab at it.  Perhaps you could name the structs using the full path to the trait, replacing `::` with `__` ?

**Comment by discosultan** [comment]
I gave it a go in https://github.com/asomers/mockall/pull/601.

**Comment by asomers** [comment]
Fixed by #601
